src/main/java/ru/yandex/practicum/filmorate/storages/UserDbStorage.java

```diff
@@ -0,0 +1,163 @@
+package ru.yandex.practicum.filmorate.storages;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+import org.springframework.stereotype.Component;
+import ru.yandex.practicum.filmorate.exceptions.UserException;
+import ru.yandex.practicum.filmorate.model.FriendshipStatus;
+import ru.yandex.practicum.filmorate.model.User;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class UserDbStorage implements UserStorage {
+    private final JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Collection<User> findAll() {
+        final String sqlQuery = "select * from users";
+        return jdbcTemplate.query(sqlQuery, this::createUser);
+    }
+
+    private User createUser(ResultSet resultSet, int rowNum) throws SQLException {
+        final int id = resultSet.getInt("id");
+        final String email = resultSet.getString("email");
+        final String login = resultSet.getString("login");
+        final String name = resultSet.getString("name");
+        final LocalDate birthday = resultSet.getDate("birthday").toLocalDate();
+        User user = new User(id, email, login, name, birthday);
```

Comment:
Можно сразу возвращать нового пользователя, не создавая для этого переменную:
```suggestion
        return new User(id, email, login, name, birthday);
```